pytorch_lightning/metrics/functional/psnr.py

```diff
@@ -1,28 +1,36 @@
-from typing import Optional, Tuple
+from typing import Optional, Tuple, Union
 
 import torch
 
-from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.metrics import utils
 
 
-def _psnr_compute(
-    sum_squared_error: torch.Tensor,
-    n_obs: int,
+def _psnr(
+    mean_squared_error: torch.Tensor,
     data_range: float,
     base: float = 10.0,
-    reduction: str = 'elementwise_mean',
 ) -> torch.Tensor:
-    if reduction != 'elementwise_mean':
-        rank_zero_warn(f'The `reduction={reduction}` parameter is unused and will not have any effect.')
-    psnr_base_e = 2 * torch.log(data_range) - torch.log(sum_squared_error / n_obs)
+    psnr_base_e = 2 * torch.log(data_range) - torch.log(mean_squared_error)
     psnr = psnr_base_e * (10 / torch.log(torch.tensor(base)))
     return psnr
 
 
-def _psnr_update(preds: torch.Tensor, target: torch.Tensor) -> Tuple[torch.Tensor, int]:
-    sum_squared_error = torch.sum(torch.pow(preds - target, 2))
-    n_obs = target.numel()
-    return sum_squared_error, n_obs
+def _psnr_compute(psnrs: torch.Tensor, reduction: str = 'elementwise_mean') -> torch.Tensor:
+    return utils.reduce(psnrs, reduction)
```

Comment:
does this mean that `reduction` will now have an effect?

Reply:
Yes. The `PSNR` uses `reduction` as [`SSIM`](https://github.com/PyTorchLightning/pytorch-lightning/blob/b2950296d59e825486c3e7a4c5b6e2b27f0c9965/pytorch_lightning/metrics/functional/ssim.py#L112) does.
https://github.com/PyTorchLightning/pytorch-lightning/blob/b2950296d59e825486c3e7a4c5b6e2b27f0c9965/pytorch_lightning/metrics/functional/ssim.py#L112